### PR TITLE
Add file-type check to filter out non-images

### LINKED ISSUE #5: Detect returned content type
Ensure returned items are of image type.

### PR DESCRIPTION
**Pull request by sallar**
Fixes #5

Todo:
- [x] Ensure the found `Buffer` has a type
- [x] Add a whitelist of mime-types

### CODE REVIEW

File: source/fetching.js [hunk 1]
```diff
@@ -1,8 +1,10 @@
+const fileType = require("file-type");
 const { parseFragment } = require("parse5");
 const { parse: parseURL, resolve: resolveURL } = require("url");
 const { getDataFetcher, getTextFetcher } = require("./fetch.js");
 
 const ICON_REL = /(apple-touch-icon|\bicon\b)/;
+const EXT_WHITELIST = ["jpg", "png", "gif", "ico", "bmp"];
```
Comment (perry-mitchell) [inline source/fetching.js]: Svg?

**Review by perry-mitchell — COMMENTED** [review]
One item.. :D

File: source/fetching.js (hunk 1 above)
Comment (sallar) [reply source/fetching.js]: Yes!

**Review by perry-mitchell — APPROVED** [review]

**Comment by perry-mitchell** [comment]
I definitely agree that we could/should return the mimetype and extension.. good idea 😊

**Pull request merged**